plugins/org.eclipse.n4js/src/org/eclipse/n4js/validation/validators/N4JSFlowgraphValidator.java

```diff
@@ -0,0 +1,251 @@
+		return null;
+	}
+
+	/**
+	 * Checks all flow graph related validations
+	 */
+	@Check
+	public void checkFlowGraphs(Script script) {
+		// Note: The Flow Graph is NOT stored in the meta info cache. Hence, it is created here at use site.
+		// In case the its creation is moved to the N4JSPostProcessor, care about an increase in memory consumption.
+		N4JSFlowAnalyser flowAnalyzer = new N4JSFlowAnalyser(this::checkCancelled);
+
+		DeadCodeAnalyser dcv = new DeadCodeAnalyser();
+		UsedBeforeDeclaredAnalyser cvgv1 = new UsedBeforeDeclaredAnalyser();
+		NullDereferenceAnalyser nda = new NullDereferenceAnalyser();
+
+		flowAnalyzer.createGraphs(script);
+		flowAnalyzer.accept(dcv, nda, cvgv1);
+
+		String uriString = script.eResource().getURI().toString();
+		Measurement msmnt1 = dcFlowGraphs.getMeasurement("flowGraphs_" + uriString);
+		Measurement msmnt2 = dcPostprocessing.getMeasurement("createGraph_" + uriString);
+		internalCheckDeadCode(dcv);
+		internalCheckUsedBeforeDeclared(cvgv1);
+		internalCheckNullDereference(nda);
+		msmnt2.end();
+		msmnt1.end();
+	}
+
+	// Req.107
+	private void internalCheckDeadCode(DeadCodeAnalyser dcf) {
+		Set<DeadCodeRegion> deadCodeRegions = dcf.getDeadCodeRegions();
+
+		for (DeadCodeRegion deadCodeRegion : deadCodeRegions) {
+			String stmtDescription = getStatementDescription(deadCodeRegion);
+			String errCode = IssueCodes.FUN_DEAD_CODE;
+			String msg = IssueCodes.getMessageForFUN_DEAD_CODE();
+			if (stmtDescription != null) {
+				msg = IssueCodes.getMessageForFUN_DEAD_CODE_WITH_PREDECESSOR(stmtDescription);
+				errCode = IssueCodes.FUN_DEAD_CODE_WITH_PREDECESSOR;
+			}
+			addIssue(msg, deadCodeRegion.getContainer(), deadCodeRegion.getOffset(), deadCodeRegion.getLength(),
+					errCode);
+		}
+	}
+
+	private void internalCheckUsedBeforeDeclared(UsedBeforeDeclaredAnalyser ubda) {
+		List<IdentifierRef> usedBeforeDeclared = ubda.getUsedButNotDeclaredIdentifierRefs();
+
+		for (IdentifierRef idRef : usedBeforeDeclared) {
+			String varName = idRef.getId().getName();
+			String msg = IssueCodes.getMessageForCFG_USED_BEFORE_DECLARED(varName);
+			addIssue(msg, idRef, IssueCodes.CFG_USED_BEFORE_DECLARED);
+		}
+	}
+
+	private String getStatementDescription(DeadCodeRegion deadCodeRegion) {
+		ControlFlowElement reachablePred = deadCodeRegion.getReachablePredecessor();
+		if (reachablePred == null)
+			return null;
+
+		String keyword = keywordProvider.keyword(reachablePred);
+		if (Strings.isNullOrEmpty(keyword)) {
+			return reachablePred.eClass().getName();
+		}
+		return keyword;
+	}
+
+	private void internalCheckNullDereference(NullDereferenceAnalyser nda) {
+		Iterable<NullDereferenceResult> nullDerefs = nda.getNullDereferences();
+		for (NullDereferenceResult ndr : nullDerefs) {
+			String varName = ndr.checkedSymbol.getName();
+
+			boolean isLeakingToClosure = isLeakingToClosure(ndr);
+			boolean isInTestFolder = isInTestFolder(ndr.checkedSymbol.getASTLocation());
```

Comment:
Is there anyway to ignore this validation in most Xpect tests except in those tests that test this null dereference?